### PR TITLE
Uploading preview adjustments

### PR DESCRIPTION
**Pull request by flamerohr**
fixes silverstripe/silverstripe-asset-admin#221

Also fixes:
* preview image size (was always blown up)
* overlay "view" showing for upload preview, which doesn't do anything.
* Added missing unit tests (incomplete, missing image loading unit test)
* `handleAddedFile` returns a promise, so it can be chained for unit testing (and if the caller wants to chain something else)

Found new issue `Gallery.handleCancelUpload()` is broken

### CODE REVIEW

File: client/src/components/GalleryItem/GalleryItem.js [hunk 1]
```diff
@@ -154,13 +154,17 @@ class GalleryItem extends SilverStripeComponent {
     return this.props.item.exists;
   }
 
+  uploading() {
```
Comment (chillu) [inline client/src/components/GalleryItem/GalleryItem.js]: Masking a prop in a trivial getter seems unnecessary here - it just makes the code harder to read. What's the rationale for this?

Comment (flamerohr) [inline client/src/components/GalleryItem/GalleryItem.js]: The intent was that the `exists()` function was already in place and I was originally tweaking that function instead.
Turned out it needed splitting out and I had found it easier to read `this.uploading()` alongside `this.exists()` as opposed to mixing `this.props.uploading` with `this.exists()`

Comment (chillu) [inline client/src/components/GalleryItem/GalleryItem.js]: Yep, makes sense

**Comment by chillu** [comment]
> Uploading preview adjustments (fixes silverstripe/silverstripe-asset-admin#221)

Feedback on commit message: You can leave out the URL if its on the same repo, so this should be:

> Uploading preview adjustments (fixes #221)

Also, careful with line length on commit message first lines - it should be less than 100 chars or so, otherwise most git repo browsers (like github) will cut it

**Comment by chillu** [comment]
Rebased on master and force pushed.

**Pull request merged**